source/modulo_components/modulo_components/lifecycle_component.py

```diff
@@ -218,7 +226,10 @@ def __handle_deactivate(self) -> bool:
         """
         result = self.__deactivate_outputs()
         try:
-            return result and self.on_deactivate_callback()
+            cb_result = self.on_deactivate_callback()
+            if cb_result is None:
+                self.get_logger().error("'on_deactivate_callback' doesn't return a value, transition will be rejected.")
```

Comment:
```suggestion
                self.get_logger().error("'on_deactivate_callback' doesn't return a value, transition will be rejected.")
                return False
```
unless I am missing something, otherwise we end up doing:

```python
return None and <some Truthy/Falsy expression> # which returns None

Reply:
Aaah very good catch thanks!